music21/musicxml/m21ToXml.py

```diff
@@ -3508,8 +3508,8 @@ def noteToXml(self, n: note.GeneralNote, noteIndexInChord=0, chordParent=None):
             chordOrN = n
         else:
             chordOrN = chordParent
-            # Ensure color is read from `n`, since only `chordOrN` is handled below
-            self.setColor(mxNote, n)
+            # Ensure style is read from `n` before reading from `chordOrN`
+            self.setPrintStyle(mxNote, n)  # sets color
```

Comment:
Thus, anything that should be overwritten by the chord's style, e.g. stem style, will be.